programs/fileio.c

```diff
@@ -2536,10 +2563,31 @@ FIO_determineDstName(const char* srcFileName, const char* outDirName)
     /* note : dstFileNameBuffer memory is not going to be free */
 }
 
+static const char*
+FIO_determineMirroredDstName(const char* srcFileName, const char* outDirName)
```

Comment:
So there are 2 functions, one called `FIO_determineMirroredDestName()`, and the other `FIO_determineMirroredDstName()`, doing essentially the same thing ?
If they do, they should be merged.
If they do not, the difference between them should be expressed in a more explicit way.

Reply:
yeah... I tried to merge it. The issue is each need call a static function which also has macro guard. 

I agree it is kinda ugly duplicated code here. Let me see if I can take another try to merge them.

Reply:
I see, there is indeed one function invocation which is different.

Sometimes, I try to solve such issue by making the invoked function an argument, but that requires both functions to have the same prototype, and that's not the case here. This can be solved with a "shell" prototype, which will just be used to present the same interface, discard unused arguments and then invoke the "real" function.

Anyway, the issue is actually less about code duplication than it is about naming. The 2 function names are just too close, they do not describe what is the difference between them, hence why they both need to exist.

Reply:
I put too much implementation details here in fileio.c, all these implementation should be only exist in the util.c. I export only 2 API nows in coming PR:
```
void  UTIL_mirrorSourceFilesDirectories(const char** fileNamesTable, unsigned int nbFiles, const char *outDirName);
char* UTIL_createMirroredDestDirName(const char* srcFileName, const char* outDirRootName);
```
 It seems reduce the need for me to write a dedicated function.